docs/cypress/support/commands.ts

```diff
@@ -36,41 +36,109 @@
-    cy.wait(3000);
-    cy.get('[data-test=Docs-Toc--Link]').each((page) => {
-        cy.get('[data-test=DesignSystem-Heading]').contains(`${page.text()}`);
-    });
+  cy.wait(3000);
+  cy.get('[data-test=Docs-Toc--Link]').each((page) => {
+    if (page.text().length) {
+      cy.request(page.prop('href'));
+      cy.get('[data-test=DesignSystem-Heading]').contains(`${page.text()}`);
+    }
+  });
 });
 
 Cypress.Commands.add('imageRender', () => {
-    cy.viewport('macbook-15');
-    cy.wait(3000)
-    cy.get('img').each(($img) => {
-        cy.wrap($img).scrollIntoView().should('be.visible').should('have.attr', 'alt');
-    });
+  cy.viewport('macbook-15');
+  cy.wait(3000);
+  cy.get('img').each(($img) => {
+    cy.wrap($img).scrollIntoView().should('be.visible').should('have.attr', 'alt');
+  });
 });
 
 Cypress.Commands.add('tileToggle', () => {
-    cy.wait(1000);
-    cy.get('.Tile').click({
-        multiple: true,
-        force: true
-    });
+  cy.wait(1000);
+  cy.get('.Tile').click({
+    multiple: true,
+    force: true,
+  });
 });
 
 Cypress.Commands.add('linkVisit', () => {
-    cy.wait(1000);
-    cy.get('[data-test=Docs-inner--container]').find('a').each((page) => {
+  cy.wait(1000);
+  cy.get('[data-test=Docs-inner--container]')
+    .find('a')
+    .each((page) => {
+      if (page.prop('href').length) {
         cy.request(page.prop('href'));
-    })
-})
+      }
+    });
+});
 
 Cypress.Commands.add('leftnavTraverse', (arr) => {
-    cy.get('[data-test=Docs-Leftnav]').find('a').not(':contains("started")').each(page => {
+  cy.get('[data-test=Docs-Leftnav]')
+    .find('a')
+    .not(':contains("started")')
+    .each((page) => {
+      cy.request(page.prop('href'));
+    });
+  cy.get('[data-test=DesignSystem-VerticalNav--Item]').each((navLink) => {
+    arr.push(navLink.prop('href'));
+  });
+});
+
+Cypress.Commands.add('livePreview', () => {
+  cy.get('[data-test=Docs-Main--Row]').then((page) => {
+    if (page.find('[data-test=Docs-live--provider]').length) {
+      cy.wait(3000);
+      cy.get('[data-test=Docs-live--provider]').find('[data-test=Docs-error-message]').should('not.exist');
+    }
+    cy.get('[data-test=Docs-inner--container]').find('[data-test=Docs-propTable-error]').should('not.exist');
+  });
+});
+
+const fetchPath = (tabName: string, url: string) => {
+  const tabSlug = tabName.toLowerCase().replace(/\s/g, '-');
+  const pagePath = url.split('/');
+  const pages = pagePath.slice(0, pagePath.length - 2);
+  const path = `${pages.join('/')}/${tabSlug}/`;
+  return path;
+}
+
+Cypress.Commands.add('tabsVisit', () => {
+  cy.get('[data-test=Docs-Main--Row]').then(($body) => {
+    if ($body.find('[data-test=DesignSystem-Tabs--Tab]').length > 1) {
+      cy.get('[data-test=DesignSystem-Tabs--Tab]')
+        .next()
+        .each((page) => {
+          const path = fetchPath(page[0].innerText, page[0].baseURI);
+          cy.visit(path);
+          if (path.includes('code')) {
```

Comment:
Why we are restricting this test only to the `code` page when the `usage` tab also contains the live preview?

Reply:
Because, we are already visiting usage page while traversing left navigation pages.